packages/constellation/src/components/Base/Padding/Padding.types.ts

```diff
@@ -0,0 +1,52 @@
+import { CSSProperties, FC } from 'react'
+
+export type PaddingVertical =
+  | {
+      all?: never
+      bottom?: never
+      top?: never
+      vertical?: number
+    }
+  | {
+      all?: never
+      bottom?: number
+      top?: number
+      vertical?: never
+    }
+  | {
+      all?: number
+      bottom?: never
+      top?: never
+      vertical?: never
+    }
+
+export type PaddingHorizontal =
+  | {
+      all?: never
+      horizontal?: number
+      left?: never
+      right?: never
+    }
+  | {
+      all?: never
+      horizontal?: never
+      left?: number
+      right?: number
+    }
+  | {
+      all?: number
+      horizontal?: never
+      left?: never
+      right?: never
+    }
+
+export type PaddingBaseProps = {
+  style?: Omit<
+    CSSProperties,
+    'padding' | 'paddingTop' | 'paddingBottom' | 'paddingLeft' | 'paddingRight'
+  >
```

Comment:
Same question as above. Beyond Blockchain's Content Security Policy for inline styles, I'm also concerned that the style prop could get misused/abused in terms of non-standard styling that deviates from the standards set by the styleguide + component library.

Reply:
Yeah on most components the style prop has the potential to be abused and cause problems but I think we are at least a little safer using it on these layout components. Being able to apply styles to these components should mean they don't have to be wrapped with another styles div. I am going to add a className prop now but I think we will have to keep `style` to keep it consistent with the previous versions

Reply:
Ok, it makes sense to me if this is to maintain the previous component API. Maybe I misunderstood the concern about Blockchain's CSP and inline styles. @blockdylanb are you able to clarify -- are inline styles ok to use?

Reply:
I just did a quick look at all the uses of Padding in the wallet, and we never pass any custom styles to it. Which, in my opinion is better, so let's just remove the ability to pass custom styles. It shouldn't break anything.